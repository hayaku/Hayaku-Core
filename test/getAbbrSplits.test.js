// Proxy for jest test to write abbr only once, ok
const test = (desc, func) => it(desc, () => func(desc));
test.only = (desc, func) => it.only(desc, () => func(desc));
test.skip = (desc, func) => it.skip(desc, () => func(desc));

const getAbbrSplits = require('./../lib/getAbbrSplits.js');

test('p', abbr => expect(getAbbrSplits(abbr)).toMatchSnapshot());
test('w20', abbr => expect(getAbbrSplits(abbr)).toMatchSnapshot());
test('poa:', abbr => expect(getAbbrSplits(abbr)).toMatchSnapshot());
test('p5:5', abbr => expect(getAbbrSplits(abbr)).toMatchSnapshot());
test('p:5:5:5', abbr => expect(getAbbrSplits(abbr)).toMatchSnapshot());
test('f:l', abbr => expect(getAbbrSplits(abbr)).toMatchSnapshot());
