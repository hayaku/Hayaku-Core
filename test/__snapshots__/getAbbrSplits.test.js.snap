// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`f:l 1`] = `
Array [
  Object {
    "prop": "f",
    "raw": "f:l",
    "val": "l",
  },
]
`;

exports[`p 1`] = `
Array [
  Object {
    "prop": "p",
    "raw": "p",
    "val": "",
  },
]
`;

exports[`p:5:5:5 1`] = `
Array [
  Object {
    "prop": "p",
    "raw": "p:5:5:5",
    "val": "5:5:5",
  },
]
`;

exports[`p5:5 1`] = `
Array [
  Object {
    "prop": "p",
    "raw": "p5:5",
    "val": "5:5",
  },
]
`;

exports[`poa: 1`] = `
Array [
  Object {
    "prop": "poa",
    "raw": "poa:",
    "val": "",
  },
]
`;

exports[`w20 1`] = `
Array [
  Object {
    "prop": "w",
    "raw": "w20",
    "val": "20",
  },
]
`;
